src/credentials/attestation_verifier/DeviceAttestationDelegate.h

```diff
@@ -0,0 +1,60 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <credentials/attestation_verifier/DeviceAttestationVerifier.h>
+#include <lib/core/Optional.h>
+
+namespace chip {
+
+namespace Controller {
+class DeviceCommissioner;
+} // namespace Controller
+
+namespace Credentials {
+
+/// Callbacks for CHIP device attestation status
+class DeviceAttestationDelegate
+{
+public:
+    virtual ~DeviceAttestationDelegate() {}
+
+    /**
+     * @brief
+     *   If valid, value to set for the fail-safe timer before the delegate's OnDeviceAttestionFailed is invoked.
+     *
+     * @return Optional value for the fail-safe timer in seconds.
+     */
+    virtual Optional<uint16_t> FailSafeExpiryTimeoutSecs() const = 0;
+
+    /**
+     * @brief
+     *   This method is invoked when device attestation fails for a device that is being commissioned. The client
+     *   handling the failure has the option to continue commissionning or fail the operation.
+     *
+     *   @param deviceCommissioner The commissioner object that is commissioning the device
+     *   @param remoteNodeId The NodeId of the Commissionee device
+     *   @param attestationResult The failure code for the device attestation validation operation
+     */
+    virtual void OnDeviceAttestionFailed(chip::Controller::DeviceCommissioner * deviceCommissioner, chip::NodeId remoteNodeId,
```

Comment:
The commissionee won't have credentials yet, so won't have a node ID unless the implementation is one where the local commissioner is assigning IDs.

Wouldn't it make more sense to pass back the CommissioneeDeviceProxy to the DeviceAttestationDelegate?

Reply:
The "remoteNodeId" here is the `remoteDeviceId` arg that was passed to `Commission`.

We need to sort out the typing of these args....  But at the very least we should clearly document that that's what the argument is.

We _could_ pass the `CommissioneeDeviceProxy *`; they're equivalent in the sense that you can get the device id from that, or you can call `GetDeviceBeingCommissioned` to get a `CommissioneeDeviceProxy` from a "NodeId" (which is not really a node id there).

Reply:
Currently, the remoteNodeId is the one from 

 NodeId remoteNodeId = commissioner->mDeviceBeingCommissioned->GetDeviceId();

which is the same as the DeviceProxy *. 

@msandstedt Do you prefer switching to DeviceProxy * proxy?

Reply:
Passing remoteNodeID here and that making sense is predicated on the commissioner being one that specifies node IDs _to_ the signer and on the commissioner being on the fabric the commissionee is being commissioned to.  Under these circumstances, yes this is the node ID that will eventually end up in the commissionee's NOC, and fabric scope will be that of the commissioner.

But if your signer assigns node IDs, remoteNodeId has no meaning.  And if the commissioner is not actually on the fabric, then fabric scope for the node ID isn't necessarily obvious or knowable.

So yes, DeviceProxy *, CommissioneeDeviceProxy *, or similar would be better because something like that is meaningful and sensible for all implementations.  As it sits, this PR has ecosystem-specific assumptions that are leaking out of the delegate implementation and into common code.

> We could pass the CommissioneeDeviceProxy *; they're equivalent in the sense that you can get the device id from that, or you can call GetDeviceBeingCommissioned to get a CommissioneeDeviceProxy from a "NodeId" (which is not really a node id there).

The `ExampleOperationalCredentialsIssuer` _does_ make it such that remoteNodeID is a real node ID because that implementation has the commissioner specifying node IDs to it, the signer, such that they end up in the signed NOCs.  But this is only one possible pattern.  We shouldn't propagate this pattern in common code.

Reply:
@msandstedt Updated to use DeviceProxy*